### ISSUE TITLE
when fuzzy-finding, grouped repositories is a bit confusing

### ISSUE THREAD

**Issue by shiftkey** [desktop/desktop#4860]
**Is your feature request related to a problem? Please describe.**
@nickcraver tweeted me this screenshot a few days ago:

![](https://user-images.githubusercontent.com/359239/40893732-efa9693c-67e2-11e8-9365-676d5590e7c9.png)

Note that `Chat` - the project that Nick wants and the best match - is actually below other matches  because grouping still takes precedence over the fuzzy-find results when we're filtering.

**Describe the solution you'd like**
I'm not settled on a solution for this, but random thoughts:

 - once fuzzy-finding kicks in, maybe we drop the grouping of repositories? But that throws away the grouping which might confuse people in a different way.
 - can we rework this UI so that we don't do the normal grouping, but still keep that context around?

**Describe alternatives you've considered**
Not really, just writing this up to capture it.

**Teachability, Documentation, Adoption, Migration Strategy**
 🤷‍♂️

**Comment by j-f1** [comment]
How about ordering the groups based on the weight of the best match in the group?

**Comment by shiftkey** [comment]
> How about ordering the groups based on the weight of the best match in the group?

Maybe? I think we'd need a spike branch with some dummy repository/account data to experiment with from a UX perspective, to figure out what makes sense.

**Comment by NickCraver** [comment]
How many groupings are there? If it's only GitHub/Enterprise, perhaps they could have their own icon, or color? If those below them shared the same, removing grouping headers during search wouldn't throw the user off, maybe.

Just a thought, if that helps!

**Comment by shiftkey** [comment]
> How many groupings are there?

Potentially three - GitHub.com, GitHub Enterprise, and Other

>  If it's only GitHub/Enterprise, perhaps they could have their own icon, or color?

We already have different icons for private repositories, forked repositories and other:

<img width="177" src="https://user-images.githubusercontent.com/359239/40894123-7ddaaf4c-67e6-11e8-907c-fbf96cec797b.png">

I'm mostly musing at this point, haven't thought too deeply about it either.

**Comment by outofambit** [comment]
While the UI underlying this issue has changed somewhat in #6460, this might still be an underlying issue with fuzzy finding and grouping in the repository list.

However, there's not been activity in this issue for almost a year, so i'm going to close it with the understanding we may reopen to focus on this in the future.

**Comment by outofambit** [comment]
https://github.com/desktop/desktop/issues/7769 is another example of this problem.

reopening this to continue discussion

**Comment by vinkla** [comment]
I agree with #7769 that the new filter updates makes it harder to jump between repositories.

1. When I filter the list I would expect it to sort exact matches before repositories in the recent group.
2. When I press <kbd>Enter</kbd> I would expect it to jump to the first repository in the list.

## Example

In this case I want to jump to the `atom-ide-definitions` repository. Though, I've recently worked in the `autocomplete-tailwind` repository. When I filter "atom" it returns `autocomplete-tailwind` before `atom-ide-definitions`. I would expect it to list the `atom-*` repositories first.  


<img src="https://user-images.githubusercontent.com/499192/60126672-ca7f4780-978e-11e9-86fb-db401edd3c08.png" width="400px">
